assets/js/googlesitekit/modules/index.js

```diff
@@ -30,12 +30,59 @@ import './datastore';
 
 const Modules = {
 	createModuleStore,
+	/**
+	 * Activates Module
+	 *
+	 * Activates a module in the modules store based on the provided args.slug parameter.
+	 *
+	 * @since 1.8.0
+	 *
+	 * @param {Object} [args]      Module arguments.
+	 * @param {string} [args.slug] Slug of the module to activate.
+	 * @return {Object} Object with `{response, error}`. On success, `response.moduleReauthURL`
+	 *                  is set to redirect the user to the corresponding module setup or OAuth
+	 *                  consent screen.
+	 */
 	activateModule: ( ...args ) => {
 		return Data.dispatch( STORE_NAME ).activateModule( ...args );
 	},
+	/**
+	 * Deactivates Module
+	 *
+	 * Deactivates a module in the modules store based on the provided args.slug parameter.
+	 *
+	 * @since 1.8.0
+	 *
+	 * @param {Object} [args]      Module arguments.
+	 * @param {string} [args.slug] Slug of the module to activate.
+	 * @return {Object} Object with `{response, error}`.
+	 */
 	deactivateModule: ( ...args ) => {
 		return Data.dispatch( STORE_NAME ).deactivateModule( ...args );
 	},
+	/**
+	 * Registers Module
+	 *
+	 * Registers a module in the modules store based on the provided args.slug and args.settings parameters.
+	 *
+	 * @since 1.13.0
+	 * @since 1.20.0 Introduced the ability to register settings and setup components.
+	 * @since 1.22.0 Introduced the ability to add a checkRequirements function.
+	 *
+	 * @param {Object}      [args]                                Module arguments.
+	 * @param {string}      [args.slug]                           Module slug.
+	 * @param {Object}      [args.settings]                       Optional. Module settings.
+	 * @param {string}      [args.settings.name]                  Optional. Module name. Default is the slug.
+	 * @param {string}      [args.settings.description]           Optional. Module description. Default empty string.
+	 * @param {string}      [args.settings.icon]                  Optional. Module icon. Default empty string.
+	 * @param {number}      [args.settings.order]                 Optional. Numeric indicator for module order. Default 10.
+	 * @param {string}      [args.settings.homepage]              Optional. Module homepage URL. Default empty string.
+	 * @param {WPComponent} [args.settings.settingsEditComponent] Optional. React component to render the settings edit panel. Default none.
+	 * @param {WPComponent} [args.settings.settingsViewComponent] Optional. React component to render the settings view panel. Default none.
+	 * @param {WPComponent} [args.settings.setupComponent]        Optional. React component to render the setup panel. Default none.
+	 * @param {Function}    [args.settings.checkRequirements]     Optional. Function to check requirements for the module. Throws a WP error object for error or returns on success.
+	 * @return {Function}   Generator Function with updated instance of the module registry.
```

Comment:
In contrast with the underlying action creator, the return value of a dispatched action (bound action creator) is always a `Promise`.
See https://github.com/WordPress/gutenberg/tree/master/packages/data#dispatch